realm-elasticsearch/lib/realm/elasticsearch/repository.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'yaml'
+require 'active_support/core_ext/string'
+
+module Realm
+  module Elasticsearch
+    class Repository
+      def self.repo_name(value = :not_provided)
+        @repo_name = value.to_sym unless value == :not_provided
+        @repo_name = name.demodulize.underscore unless defined?(@repo_name)
+        @repo_name
+      end
+
+      def self.index_name(value = :not_provided)
+        @index_name = value.to_sym unless value == :not_provided
+        @index_name = repo_name.pluralize unless defined?(@index_name)
+        @index_name
+      end
+
+      def initialize(client)
+        @client = client
+      end
+
+      def find(id:)
+        single(client.get(index: index_name, id: id))
+      rescue ::Elasticsearch::Transport::Transport::Errors::NotFound
+        nil
```

Comment:
Anti-pattern when rescuing without any result. This is dangerous because you can hide some errors.

Reply:
As above